### DISCUSSION TITLE
Inconsistencies between the API and generated playlists

### DISCUSSION THREAD

**Discussion by jacklul** [iptv-org/community#508] [Q&A]
### Describe your issue

Apologizes if I opened the issue in the wrong place, there is nothing about such issues in the Contributing Guide. This issue is related to the scripts in this repository so I opened it here.

I believe API has some issues as well as scripts that generate playlists.

The issues:

https://iptv-org.github.io/?q=languages:pol
Why are non Polish channels displayed here?

https://iptv-org.github.io/iptv/languages/pol.m3u
Why are non Polish language channels here?

```
#EXTINF:-1 tvg-id="BabyFirst.us" tvg-logo="https://i.imgur.com/WxLHBwu.png" group-title="Kids",BabyFirst
#EXTINF:-1 tvg-id="EbS.be" tvg-logo="https://i.imgur.com/DIIRPNN.png" group-title="Legislative",EbS Live (Europe by Satellite) (1080p)
#EXTINF:-1 tvg-id="EbSPlus.be" tvg-logo="https://i.imgur.com/7V012zQ.png" group-title="Legislative",EbS+ Live (Europe by Satellite) (1080p)
```

https://iptv-org.github.io/iptv/countries/pl.m3u
Why are non Poland hosted channels here?

API does not even list them, so where did they came from? https://iptv-org.github.io/?q=country:pl

```
#EXTINF:-1 tvg-id="AngelTVEurope.in" tvg-logo="https://i.imgur.com/qKLEGU7.png" group-title="Religious",Angel TV Europe (720p)
#EXTINF:-1 tvg-id="BloombergTVEMEALiveEvent.uk" tvg-logo="https://i.imgur.com/OuogLHx.png" group-title="Business",Bloomberg TV EMEA Live Event (720p)
#EXTINF:-1 tvg-id="BloombergTVEurope.uk" tvg-logo="https://i.imgur.com/OuogLHx.png" group-title="Business",Bloomberg TV Europe (720p)
#EXTINF:-1 tvg-id="EbS.be" tvg-logo="https://i.imgur.com/DIIRPNN.png" group-title="Legislative",EbS Live (Europe by Satellite) (1080p)
#EXTINF:-1 tvg-id="EbSPlus.be" tvg-logo="https://i.imgur.com/7V012zQ.png" group-title="Legislative",EbS+ Live (Europe by Satellite) (1080p)
#EXTINF:-1 tvg-id="EWTNEurope.us" tvg-logo="https://upload.wikimedia.org/wikipedia/en/thumb/c/c8/EWTN_Logo_and_Wordmark_%282016%29.svg/512px-EWTN_Logo_and_Wordmark_%282016%29.svg.png" group-title="Religious",EWTN Europe (720p) [Not 24/7]
#EXTINF:-1 tvg-id="FashionTVEurope.fr" tvg-logo="https://upload.wikimedia.org/wikipedia/commons/thumb/7/7a/Fashion_TV_logo.svg/512px-Fashion_TV_logo.svg.png" group-title="Lifestyle",Fashion TV (1080p)
#EXTINF:-1 tvg-id="Telekriti.gr" tvg-logo="https://telekriti.com/wp-content/uploads/2023/08/NEW-LOGO-7c-300x50.png" group-title="General",Telekriti (720p) [Not 24/7]
#EXTINF:-1 tvg-id="TV5MondeEurope.fr" tvg-logo="https://i.imgur.com/rPAbMSu.png" group-title="General",TV5Monde Europe (1080p) [Geo-blocked]
```

### Contributing Guide

- [X] I have read [Contributing Guide](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md)

**Comment by Dum4G** [comment]
Poland is considered EUR region so have those channels slapped into Polish playlist as if they were broadcasted for you. You can find those channels even in Russian playlist

**Comment by BellezaEmporium** [comment]
Not a bug per-se. Some channels that come from outside Poland can be streamed to you if they're aimed at an international/european audience. Those channels are, therefore, included in the Poland playlist.

**Comment by jacklul** [comment]
> Poland is considered EUR region so have those channels slapped into Polish playlist as if they were broadcasted for you. You can find those channels even in Russian playlist

What is the point of having a playlist that lists only channels in specific language when it also has foreign language ones?

> Not a bug per-se. Some channels that come from outside Poland can be streamed to you if they're aimed at an international/european audience. Those channels are, therefore, included in the Poland playlist.

Makes sense, though I think for this specific user case providing a diff playlist for filtering those out would be good.

I would say country lists should contain those extra channels but language ones should not.

**Reply by freearhey** [reply]
> What is the point of having a playlist that lists only channels in specific language when it also has foreign language ones?

This happens because the wrong language is specified in the channel description in the [iptv-org/database](https://github.com/iptv-org/database).  You can correct this by sending a request through this [form](https://github.com/iptv-org/database/issues/new?assignees=&labels=channels%3Aedit&projects=&template=__channels_edit.yml&title=Edit%3A+).

**Reply by jacklul** [reply]
I went through the `channel.csv` looking up those extra channels.
The ones on [pol.m3u](https://iptv-org.github.io/iptv/languages/pol.m3u) do have `pol` language added, with other languages of EU countries, I wouldn't touch them because that setup actually makes sense.
And for [pl.m3u](https://iptv-org.github.io/iptv/countries/pl.m3u) they just have EU region set., so that confirms what @BellezaEmporium said.

So while [languages:pol](https://iptv-org.github.io/?q=languages:pol) returns results matching `pol.m3u`, results of [country:pl](https://iptv-org.github.io/?q=country:pl) does not match `pl.m3u`.

**Reply by freearhey** [reply]
`country:pl` returns only  the list of all channels broadcasting from Poland. 
To see the list of channels that are included in the `countries/pl.m3u` search by `broadcast_area` instead: https://iptv-org.github.io/?q=broadcast_area:c/pl

**Reply by jacklul** [reply]
Yeah, well not really exactly what I wanted to achieve but it's no problem.
I'll just `sed` the m3u list to get rid of those unwanted channels, or perhaps parse the API.
